### PR TITLE
add p2tr untweaked key spend implementation to motoko

### PR DESCRIPTION
**Pull request by altkdf**
This PR adds pay-to-taproot (P2TR) raw key path transactions to the motoko basic_bitcoin example (see the modified README.md for more details). P2TR raw key spend is the encoded untweaked public key. This type of address is insecure to use with more than one signer, i.e., with multisignatures, but this is not a concern for using it on the IC, since it does not support multisignatures in the first place. Note that this type of address does not support scripts.

### CODE REVIEW

**Comment by altkdf** [comment]
[This](https://github.com/altkdf/motoko-bitcoin/tree/alex/add-p2tr-key-path-experimental) is the repository and branch with the `motoko-bitcoin` implementation for P2TR untweaked key spend.

**Comment by altkdf** [comment]
Wanted to test this on testnet but testnet is currently unusable because the amount of satoshis one gets from the faucet is much smaller than the required fee, see [this](https://blockstream.info/testnet/address/tb1pzpyg5w2sttcvtvm6gj8mn6ksl5aafpu9355c7h05j7n0x9cz8jtqgnjjwr) where the faucet transferred 37248 satoshis with a fee of 85991.

**Review by fspreiss — APPROVED** [review]
Thanks, @altkdf! I'm not an expert on Bitcoin, but this looks good to me. As you mentioned offline, you will adapt the code to use the motoko bitcoin library published in mops.

**Comment by fspreiss** [comment]
Please also add a brief PR description once this is move out of draft status.

File: motoko/basic_bitcoin/src/basic_bitcoin/src/Utils.mo [hunk 1]
```diff
@@ -4,28 +4,29 @@ import Iter "mo:base/Iter";
...
                 value;
             case (#err error)
-                Debug.trap("pattern failed");
+                Debug.trap("pattern failed: " # error);
         }
     };
 
     /// Returns the value of the result and traps with a custom message if there isn't any value to return.
-    public func get_ok_except<T, U>(result : Result<T, U>, expect : Text) : T {
+    public func get_ok_except<T>(result : Result<T, Text>, expect : Text) : T {
```
[the remark this answers is not in the document]
Comment (altkdf) [reply motoko/basic_bitcoin/src/basic_bitcoin/src/Utils.mo]: done

**Comment by altkdf** [comment]
thanks for the review @fspreiss! I added some modifications to the README -- you may want to take a look

File: motoko/basic_bitcoin/README.md [hunk 2]
```diff
@@ -79,17 +88,48 @@ In the output above, to see the Candid Web UI for your bitcoin canister, you wou
...
+from two types of addresses:
+1. A [P2PKH address](https://en.bitcoin.it/wiki/Transaction#Pay-to-PubkeyHash)
+   using the
+   [ecdsa_public_key](https://internetcomputer.org/docs/current/references/ic-interface-spec/#ic-method-ecdsa_public_key)
+   API.
+2. A [P2TR
+   address](https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki)
+   where the funds can be spent using the raw (untweaked) internal key
+   (so-called P2TR key path spend, but untweaked). The advantage of this
+   approach compared to the one below is its significantly smaller fee per
```
Comment (fspreiss) [inline motoko/basic_bitcoin/README.md:107]: It's not fully clear to me what "the one below" means here.

**Review by fspreiss — COMMENTED** [review]
Thanks for updating the README

File: motoko/basic_bitcoin/README.md (hunk 2 above)
Comment (altkdf) [reply motoko/basic_bitcoin/README.md]: That's a copy paste. The one below should refer to script spends, which are not yet implemented in this PR. Replaced with "P2TR script spends".

**Pull request merged**